### ISSUE TITLE
2024 General Election Precinct Results

### ISSUE THREAD

**Issue by dwillis** [openelections/openelections-data-or#228]
Using [Tabula](http://tabula.technology/), OCR or whatever method you can, parse precinct-level results for the following counties. Original sources are in individual county folders in the [sources-or repository](https://github.com/openelections/openelections-sources-or/).

The goal is to create a single CSV file for each county, with the following headers:

`county`, `precinct`, `office`, `district`, `party`, `candidate`, `votes`

[Here's an example of a finished CSV file](https://github.com/openelections/openelections-data-or/blob/master/2024/counties/20181106__or__general__baker__precinct.csv).

If the county file also provides a breakdown of votes by method, include that using the following headers, where applicable:

`early_voting`, `election_day`, `provisional`, `mail`

If there are other possible vote types, include them, using a lowercase version of the vote type with underscores instead of spaces for the column name.

Include the following offices:

* Registered Voters (if available)
* Ballots Cast (if available)
* President
* U.S. House
* Secretary of State
* State Treasurer
* Attorney General
* State Senator
* State Representative

If a county provides precinct results for Write-in candidates, they should be grouped in a single row for each precinct and office with a `candidate` value of `Write-ins`.

If a county provides Under Votes or Over Votes, those should be recorded in the same way, with a single row per precinct and office with `Over Votes` and `Under Votes` as the `candidate` values.

- [x] Baker
- [x] Benton
- [x] Clackamas
- [x] Clatsop
- [x] Columbia
- [x] Coos
- [x] Crook
- [x] Curry
- [x] Deschutes
- [x] Douglas
- [x] Gilliam
- [x] Grant
- [x] Harney
- [x] Hood River
- [x] Jackson
- [x] Jefferson
- [x] Josephine
- [x] Klamath
- [x] Lake
- [x] Lane
- [x] Lincoln
- [x] Linn
- [x] Malheur
- [x] Marion
- [x] Morrow
- [x] Multnomah
- [x] Polk
- [x] Sherman
- [x] Tillamook
- [x] Umatilla
- [x] Union
- [x] Wallowa
- [x] Wasco
- [x] Washington
- [x] Wheeler
- [x] Yamhill

**Comment by andysylvester** [comment]
I will take Clatsop County.

**Comment by andysylvester** [comment]
I will take Baker County

**Comment by dwillis** [comment]
@andysylvester I've got another person pitching in on Oregon and should have an update on their progress this weekend.

**Comment by warwickmm** [comment]
It looks like some contributions are being pushed directly to `main`.  Should we be using pull requests instead (except for perhaps @dwillis's changes)?

**Comment by dwillis** [comment]
@warwickmm ideally, but I also will be taking a swing through all the contributions.

**Comment by warwickmm** [comment]
Sounds good.  I was thinking that pull requests could possibly save you some effort cleaning up mistakes (for pull requests, we notify users of errors in newly added files).

For example, in `2024/counties/20241105__or__general__clatsop__precinct.csv` I think "Registered Voters" should appear under the "office" column?
```
  ======================================================================
  FAIL: test_missing_values (data_tests.test_data.MissingValuesTest.test_missing_values) [2024/counties/20241105__or__general__clatsop__precinct.csv] (group='2024')
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/home/runner/work/openelections-data-or/openelections-data-or/data_tests/data_tests/test_data.py", line 195, in test_missing_values
      self._assertTrue(passed, f"{self} [{short_path}]", console_message, log_message)
    File "/home/runner/work/openelections-data-or/openelections-data-or/data_tests/data_tests/test_data.py", line 56, in _assertTrue
      self.assertTrue(result, console_message)
  AssertionError: False is not true : 
  
  * There are 50 rows that are missing a office:
  
  	Headers: ['\ufeffcounty', 'precinct', 'office', 'district', 'party', 'candidate', 'votes']:
  	Row 802: ['Clatsop', '101 - Astoria West', '', '', '', 'Registered Voters', '3481']
  	Row 803: ['Clatsop', '102 - Astoria East', '', '', '', 'Registered Voters', '2488']
  	Row 804: ['Clatsop', '103 - John Day/Walluski', '', '', '', 'Registered Voters', '981']
  	Row 805: ['Clatsop', '104 - Astoria South', '', '', '', 'Registered Voters', '1307']
  	Row 806: ['Clatsop', '105 - Svensen', '', '', '', 'Registered Voters', '1389']
  	Row 807: ['Clatsop', '106 - Knappa', '', '', '', 'Registered Voters', '1516']
  	Row 808: ['Clatsop', '107 - Westport/Hillcrest', '', '', '', 'Registered Voters', '532']
  	Row 809: ['Clatsop', '108 - Jewell', '', '', '', 'Registered Voters', '534']
  	Row 810: ['Clatsop', '109 - Olney', '', '', '', 'Registered Voters', '571']
  	Row 811: ['Clatsop', '110 - Stanley Acres', '', '', '', 'Registered Voters', '264']
  	[Truncated to 10 examples]
```

**Comment by andysylvester** [comment]
I will take a look at that tomorrow. As for committing to master, I am happy to create branches/merge requests, just let me know.

**Comment by andysylvester** [comment]
> @andysylvester I've got another person pitching in on Oregon and should have an update on their progress this weekend.

sounds good!

**Comment by dwillis** [comment]
> I will take a look at that tomorrow. As for committing to master, I am happy to create branches/merge requests, just let me know.

Whatever is easiest for you works for me, but PRs would be great, thanks!

**Comment by andysylvester** [comment]
@dwillis I just saw that @vrlamani made a commit for Baker County, which I mentioned earlier in this issue that I was working on. This was my primary frustration the last time I worked on this project - people doing work I already said I was going to do. If this is going to keep happening, I will move on with my life. Please check with that person and ask them to post saying what counties they are working on, so there is no wasted effort. Thanks.

**Comment by vrlamani** [comment]
Hey @andysylvester! Thank you so much for flagging—sorry about the confusion. I am new and learning! Currently, I am working on the following counties: Coos, Crook, Deschutes, Harney, Jefferson, Lake, Lincoln, Malheur, Polk, Sherman, Wallowa, Wheeler, Curry, Morrow, Washington, Marrion, Hood-River, Gilliam, Josephine, Yamhill, and Wasco. Would you like to take on the remaining counties? I would love to continue collaborating with you on this :)

**Comment by dwillis** [comment]
@andysylvester My apologies - I should have headed that off. That's on me, and @vrlamani's list leaves some counties open if you're willing:

 Benton
 Clackamas
 Columbia
 Douglas
 Grant
 Jackson
 Klamath
 Lane
 Linn
 Multnomah
 Tillamook
 Umatilla
 Union

**Comment by andysylvester** [comment]
@dwillis and @vrlamani, thanks for your replies, I will start with the bottom of the list (Umatilla and Union, to not overcommit), and will post as I complete and claim another county.

**Comment by vrlamani** [comment]
Thanks @andysylvester! We got this!

**Comment by andysylvester** [comment]
I have created a pull request for updates to Clatsop County and adding a file for Union County, @dwillis let me know how to proceed.

**Comment by dwillis** [comment]
@andysylvester looks good, thanks!

**Comment by andysylvester** [comment]
I will take Tillamook County now.

**Comment by dwillis** [comment]
I've got Jackson

**Comment by andysylvester** [comment]
Thanks! I have 5 more precincts to go with Tillamook County (poorly structured PDF file), hope to finish tomorrow, will let you know what county I pick up after that.

**Comment by dwillis** [comment]
I'm working from the top down.

**Comment by andysylvester** [comment]
Derek, I created a branch and merge request for Tillamook County, saw that I had a slight data error, fixed it, and committed it to my branch again. It does not look like it was automatically added to my merge request, should I create a new request? Let me know how you would like to handle this, thanks.

**Comment by warwickmm** [comment]
@andysylvester, are you referring to revision 5b5715b49727cbf817f7f9c744677ed295a7aac6?  It looks like that's included in pull request #231.

**Comment by andysylvester** [comment]
@warwickmm , yes, I was referring to that revision - thanks! It looks like Derek has merged the request.

**Comment by andysylvester** [comment]
Multnomah is pretty large, I am going to take Grant County if Derek is not working on it...let me know!

**Comment by dwillis** [comment]
@andysylvester Go for it. I'll take Multnomah.

**Comment by dwillis** [comment]
Hey @andysylvester - i can help on Grant if you need it, it's the last county! Thanks so much for helping with Oregon.

**Comment by andysylvester** [comment]
@dwillis - go ahead and take Grant, I have gotten distracted with some other things, I don't want to be the hold-up on getting Oregon done!
